src/app/ngrx/effects/google-analytics.effects.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import 'rxjs/add/operator/switchMap';
+import { Observable } from 'rxjs/Observable';
+import { Action } from '@ngrx/store';
+import { Actions, Effect } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
+import { Angulartics2 } from 'angulartics2';
+import { ActionTypes, GoogleAnalyticsEventAction } from '../actions';
+import { selectPodcasts, selectFilter } from '../reducers';
+import { PodcastModel, FilterModel } from '../model';
+
+@Injectable()
+export class GoogleAnalyticsEffects {
+  podcasts: PodcastModel[];
+  filter: FilterModel;
+
+  @Effect({dispatch: false})
+  fromGAEvent$: Observable<Action> = this.actions$
+    .ofType(ActionTypes.GOOGLE_ANALYTICS_EVENT)
+    .map((action: GoogleAnalyticsEventAction) => {
+      const event = {
+        action: action.payload.gaAction
+      };
+      if (action.payload.category) {
+        event['category'] = action.payload.category;
+      } else {
+        if (this.filter && this.filter.interval) {
+          event['category'] = 'Downloads/' + this.filter.interval.name;
+        }
+      }
+      if (action.payload.label) {
+        event['label'] = action.payload.label;
+      } else {
+        if (this.filter && this.filter.podcastSeriesId && this.podcasts) {
+          const podcast = this.podcasts.find(p => p.seriesId === this.filter.podcastSeriesId);
+          if (podcast) {
+            event['label'] = podcast.title;
```

Comment:
Heh, so I actually broke the google analytics for a minute because the components didn't have the podcast title from the filter anymore, which is why I decided to make this an effect and have the podcast titles handled in one place. Thinking out loud a little further on these concepts of moving this stuff out of components, I'd like to create services that listen for these kind of state changes and handles the subscribe and filter logic.

Reply:
👍